### PR TITLE
Make localtime and localtime_r aware of system timezone

### PR DESCRIPTION
**Pull request by althonos**
Hi everyone !

Just a quick PR to make `localtime_r` (and subsequently `localtime`) aware of the user-specified timezone. Currently, `localtime` and `gmtime` will always return the same value (found after bissecting libretro/RetroArch#7632).

I hope it's the right way to do things, I've been working with the SDK a bit but this is my first time messing with `newlib`'s code.

### CODE REVIEW

**Comment by althonos** [comment]
NB: This has been tested on a UTC+1 PS Vita running w/ Henkaku 3.60.

File: newlib/libc/sys/vita/lcltime_r.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    time_t localtime;
+    int rid;
+
+    TRY(sceRtcConvertTime_tToDateTime(*timep, &dt));
+    TRY(sceRtcConvertDateTimeToTick(&dt, &tick));
+    TRY(sceRtcConvertUtcToLocalTime(&tick, &tick));
+    TRY(sceRtcConvertTickToDateTime(&tick, &dt));
+    TRY(sceRtcConvertDateTimeToTime_t(&dt, &localtime));
+
+    return gmtime_r(&localtime, result);
```
Comment (d3m3vilurr) [inline newlib/libc/sys/vita/lcltime_r.c]: can you use hardtab?

File: newlib/libc/sys/vita/lcltime_r.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+#include <errno.h>
+#include <sys/time.h>
+#include <sys/types.h>
+
+#include <psp2/rtc.h>
+
+#include "vitaerror.h"
+
+#define TRY(x) {rid=x; if (rid < 0) {errno = __vita_sce_errno_to_errno(rid); return NULL;}}
```
Comment (d3m3vilurr) [inline newlib/libc/sys/vita/lcltime_r.c]: maybe we need to update `error.c` file or just use `errno = rid`

https://github.com/vitasdk/vita-headers/blob/bbaaf16/include/psp2/rtc.h#L17-L31

**Review by d3m3vilurr — CHANGES_REQUESTED** [review]

**Comment by ofry** [comment]
@althonos Could you fix these cosmetic issues?

File: newlib/libc/sys/vita/lcltime_r.c (hunk 1 above)
Comment (althonos) [reply newlib/libc/sys/vita/lcltime_r.c]: Fixed (sorry, Python habits!)

File: newlib/libc/sys/vita/lcltime_r.c (hunk 2 above)
Comment (althonos) [reply newlib/libc/sys/vita/lcltime_r.c]: `lcltime_r` is not supposed to error with anything else than `EOVERFLOW`, but most of the RTC errors would probably resolve as `EINVAL` I guess ? Anyway I went with `errno = rid` for now.

**Comment by ofry** [comment]
@d3m3vilurr Could you review this PR again?

**Review by d3m3vilurr — APPROVED** [review]

**Comment by d3m3vilurr** [comment]
@althonos @ofry Thanks!

**Pull request merged**